### PR TITLE
[IE CLDNN] Improving graph initialization step

### PR DESCRIPTION
**Pull request by amolotko**

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/src/include/pass_manager.h [hunk 1]
```diff
@@ -102,9 +102,9 @@ class graph_initializations : public base_pass {
 
 private:
     void run(program_impl& p) override;
-    void replace_nodes(program_impl& p);
-    void handle_lstm(program_impl& p);
-    void handle_dynamic_lstm(program_impl& p);
+    void handle_split_node(program_impl& p, std::shared_ptr<program_node> node);
```
Comment (vladimir-paramuzov) [inline inference-engine/thirdparty/clDNN/src/include/pass_manager.h:105]: I think the parameter type should be changed to `split_node& node`, lstm_node, etc

Comment (amolotko) [reply inference-engine/thirdparty/clDNN/src/include/pass_manager.h]: done

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp [hunk 2]
```diff
@@ -38,420 +38,376 @@ std::string get_id_string(size_t i) {
...
-                                       ":<split_output_id>\"!");
+    auto split_prim = node.typed_desc();
+    std::size_t split_num = split_prim->output_offsets.size();
+
+    std::vector<primitive_id> transformed_ids;
 
-            // get_output size and validate split primitive inputs
-            auto output_layout = node->get_output_layout();
-            auto output_layout_size = output_layout.size;
+    // create crop for each split ouptut provided
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp:55]: output

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp [hunk 3]
```diff
@@ -38,420 +38,376 @@ std::string get_id_string(size_t i) {
...
-                if (user->is_type<lstm>()) {
-                    has_lstm_children = true;
-                }
+            // primitive_id lstm_gemm_input_id = node->get_dependency(input_idx).get_primitive()->id;
+            // the line below requires an attention: get_org_primitive_id() might not be an actual id of a node
+            // (see rename method) ToDO: ensure that get_org_primitive_id() is suitable here
+            primitive_id lstm_gemm_input_id = node.get_dependency(input_idx).get_org_primitive_id();
+
+            auto lstm_gemm_node = std::make_shared<lstm_gemm>(lstm_gemm_id,
+                                                                lstm_gemm_input_id,
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp:220]: nit: align

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp [hunk 4]
```diff
@@ -38,420 +38,376 @@ std::string get_id_string(size_t i) {
...
+        if (directions == 2) {
+            // bidirectional support requires concatenations along the direction and sequence axis
+            // instead we can concatenate along the sequence axis and reshape the tensor to the account
+            // for the direction
+            size_t concatenate_len = emit_sequence ? sequence_len : 1;
+            if (emit_last_cell)
+                concatenate_len++;
+
+            tensor output_size{input_size.batch[0],
+                                static_cast<int32_t>(concatenate_len),
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp:316]: nit: align

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp [hunk 5]
```diff
@@ -38,420 +38,376 @@ std::string get_id_string(size_t i) {
...
+    auto input_id = node.get_primitive()->input.at(0);
+    auto dyn_length_id = node.dyn_length_id();
+    auto weights_id = node.weights_id();
+    auto bias_id = node.bias_id();
+    std::string suffix = "__cldnn_";
+
+    // [1] Add lstm_dynamic_input
+    auto lstm_dynamic_input_primitive =
+        std::make_shared<lstm_dynamic_input>(node_id + suffix + "input",
+                                                input_id,
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp:345]: nit: align

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp [hunk 6]
```diff
@@ -38,420 +38,376 @@ std::string get_id_string(size_t i) {
...
+                                                last_hidden_id,
+                                                last_cell_id,
+                                                init_hidden_id,
+                                                init_cell_id,
+                                                node.clip(),
+                                                node.input_forget(),
+                                                lstm_dynamic_input_primitive->output_padding);
+    auto& lstm_dynamic_timeloop_node = p.get_or_create(lstm_dynamic_timeloop_primitive);
+    p.add_connection(lstm_dynamic_input_node,
+                        lstm_dynamic_timeloop_node);  // connect dlstm_input to dlstm_timeloop
```
Comment (lznamens) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp:378]: nit: align

Comment (amolotko) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp]: done

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp (hunk 5 above)
Comment (amolotko) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp]: done

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp (hunk 4 above)
Comment (amolotko) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp]: done

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp (hunk 3 above)
Comment (amolotko) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp]: done

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp (hunk 2 above)
Comment (amolotko) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/graph_initializations.cpp]: done

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**